### PR TITLE
Help Center: refactor contact form

### PR DESCRIPTION
**Pull request by arcangelini**
#### Proposed Changes

The contact form has become difficult to manage given how much logic has been crammed in there to account for each situation. This PR removes some of the logic and puts it into custom hooks. I believe this is a step in the right direction to have a cleaner component.

#### Testing Instructions

1. Pull branch and start
2. Open up help center and attempt to submit a ticket, chat, and forum. There should be no regression.
3. More steps to come. Need to test all scenarios.

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/help-center/src/components/help-center-site-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		name: __( 'Other site', __i18n_text_domain__ ),
+		ID: 0,
+		logo: { id: '', sizes: [] as never[], url: '' },
+		URL: '',
+	} as const;
+
+	const options: ( SitePickerSite | undefined )[] = [ currentSite, otherSite ];
+
+	return (
+		<>
```
Comment (alshakero) [inline packages/help-center/src/components/help-center-site-picker.tsx]: Redundant fragment.

**Review by alshakero — CHANGES_REQUESTED** [review]
Absolutely love the direction of this. Amazing work. Thanks a lot!

Started checking scenarios and stopped at free users.

- ✅  No sites user
- ❌ Free site users (forums): sibyl articles didn't show when I clicked `Continue`, it actually submitted the form right away even though there were results and the button said "Continue".

**Comment by arcangelini** [comment]
Thanks @alshakero got this all cleaned up and tested in all versions. Should be working with Sibyl now. I also had to fix the site picker when they choose other site.

**Comment by arcangelini** [comment]
Decided to break this up to smaller chunks

1. Move the Site Picker to its own element - #76986

**Comment by arcangelini** [comment]
2. Move useFormTitle to hooks directory #77377

**Pull request closed without merging**